bundles/org.openhab.binding.freeathomesystem/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,307 @@
+		<channels>
+			<channel id="switch_channel" typeId="switch_channel_type">
+				<label>Actuator Channel</label>
+				<description>Actuator (ON/OFF) Channel for Free@Home Binding</description>
+			</channel>
+		</channels>
+	</thing-type>
+
+	<!-- Dooropener Actuator Thing Type -->
+	<thing-type id="dooropener" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Actuator</label>
+		<description>Actuator for Free@Home Binding</description>
+
+		<channels>
+			<channel id="switch_channel" typeId="switch_channel_type">
+				<label>Actuator Channel</label>
+				<description>Actuator (ON/OFF) Channel for Free@Home Binding</description>
+			</channel>
+		</channels>
+	</thing-type>
+
+	<!-- Actuator Channel Type -->
+	<channel-type id="actuator_channel_type">
+		<item-type>Switch</item-type>
+		<label>Actuator Channel</label>
+		<description>Free@Home Actuator Channel</description>
+		<category>Switch</category>
+	</channel-type>
+
+	<!--Thermostat Thing Type -->
+	<thing-type id="thermostat" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Thermostat</label>
+		<description>Thermostat for Free@Home binding</description>
+
+		<channels>
+			<channel id="thermostat_setpoint_temperature" typeId="thermostat_temp_channel_type">
+				<label>Setpoint Temperature</label>
+				<description>Setpoint Temperature of Free@Home Binding</description>
+			</channel>
+			<channel id="thermostat_measured_temperature" typeId="thermostat_temp_channel_type_ro">
+				<label>Thermostat Measured Temperature</label>
+				<description>Measured Temperature of the Free@Home Thermostat</description>
+			</channel>
+			<channel id="thermostat_heating_demand" typeId="thermostat_heatingvalue_channel_type_ro">
+				<label>Thermostat Heating Demand</label>
+				<description>Heating Demand at the Free@Home Thermostat</description>
+			</channel>
+			<channel id="thermostat_heating_active" typeId="thermostat_heatingvalue_channel_type_ro">
+				<label>Thermostat Heating Active (1/0)</label>
+				<description>Heating Active at the Free@Home Thermostat</description>
+			</channel>
+			<channel id="thermostat_state" typeId="thermostat_state_channel_type">
+				<label>Thermostat State</label>
+				<description>Current State of the Free@Home Thermostat (Eco(36), Heating (0))</description>
+			</channel>
+			<channel id="thermostat_onoff_switch" typeId="thermostat_Switch_channel_type">
+				<label>Thermostat Activation (On/Off)</label>
+				<description>Thermostat Main Activation Switch</description>
+			</channel>
+			<channel id="thermostat_eco_switch" typeId="thermostat_Switch_channel_type">
+				<label>Thermostat Eco (On/Off)</label>
+				<description>Thermostat Eco Activation Switch</description>
+			</channel>
+		</channels>
+	</thing-type>
+
+	<!--Thermostat temperature channel types -->
+	<channel-type id="thermostat_temp_channel_type">
```

Comment:
All channels should use lowerCamelCase.
See https://www.openhab.org/docs/developer/guidelines.html#java-coding-style
Same comment for all channels that dont use the above.

Reply:
Coding style fixes are done